src/main/java/org/springframework/data/couchbase/core/CouchbaseTemplateSupport.java

```diff
@@ -96,6 +96,18 @@ public void applyUpdatedCas(final Object entity, final long cas) {
 		}
 	}
 
+	public long getCas(final Object entity) {
+		final ConvertingPropertyAccessor<Object> accessor = getPropertyAccessor(entity);
+		final CouchbasePersistentEntity<?> persistentEntity = mappingContext.getRequiredPersistentEntity(entity.getClass());
+		final CouchbasePersistentProperty versionProperty = persistentEntity.getVersionProperty();
+
+		long cas = 0;
+		if (versionProperty != null) {
+			cas = (long) accessor.getProperty(versionProperty);
```

Comment:
Probably safer to have an instanceof check for `Number` as domain models may contain `BigInteger` and other numeric types.

Reply:
fixed